ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -62,41 +62,53 @@ private[dao] final class TransactionsReader(
       filter: FilterRelation,
       verbose: Boolean,
   ): Source[(Offset, GetTransactionsResponse), NotUsed] = {
-    val events: Source[EventsTable.Entry[Event], NotUsed] =
-      PaginatingAsyncStream.streamFrom((startExclusive, Option.empty[Int]), getOffset[Event]) {
-        case (prevOffset, prevNodeIndex) =>
-          val query =
-            EventsTable
-              .preparePagedGetFlatTransactions(sqlFunctions)(
-                startExclusive = prevOffset,
-                endInclusive = endInclusive,
-                filter = filter,
-                pageSize = pageSize,
-                previousEventNodeIndex = prevNodeIndex,
-              )
-              .withFetchSize(Some(pageSize))
-          val rawEventsFuture =
-            dispatcher.executeSql(dbMetrics.getFlatTransactions) { implicit connection =>
-              query.asVectorOf(EventsTable.rawFlatEventParser)
-            }
-          rawEventsFuture.flatMap(
-            rawEvents =>
-              Timed.future(
-                future = Future.traverse(rawEvents)(deserializeEntry(verbose)),
-                timer = dbMetrics.getFlatTransactions.translationTimer,
+
+    val eventsRangeF: Future[EventsRange[Long]] = dispatcher.executeSql(dbMetrics.getRowIdRange) {
+      connection =>
+        EventsRange.eventsRange(EventsRange(startExclusive, endInclusive))(connection)
+    }
+
+    def getEvents(eventsRange0: EventsRange[Long]): Source[EventsTable.Entry[Event], NotUsed] =
```

Comment:
What does `0` mean here?

Reply:
You mean `0` in `eventsRange0`? It is for the initial state, `eventsRange1` -- next state